### DISCUSSION TITLE
Optional icon prop for <SelectItem> component

### DISCUSSION THREAD

**Discussion by schmitzc** [gluestack/gluestack-ui#1248] [Feature Request]
### Description

Add an additional optional prop to `<SelectItem>` to include an icon.

### Problem Statement

My app has a dropdown that uses only icons for the item labels. I am using accessibility labels to ensure that the dropdown is still accessible.

### Proposed Solution or API

I am using the [React Native Dropdown Picker](https://hossein-zare.github.io/react-native-dropdown-picker-website/) instead of the gluestack-ui Select component. It allows you to [pass an icon as optional prop to the select item](https://hossein-zare.github.io/react-native-dropdown-picker-website/docs/advanced/icons).

### Alternatives

Is it possible to create a custom gluestack-ui <SelectItem> component that can have an icon?

**Comment by surajahmed** [comment]
@schmitzc We don't have this feature for now. We'll consider your request. Thanks for the suggestion.

**Comment by Viraj-10** [comment]
This enhancement require custom select on web for design consistency. Currently we are relying on native select on web which doesn't allow us to customised `SelectItem`.

**Comment by MarlonAEC** [comment]
Is there any workaround to achieve this? For example, creating your custom SelectItem and passing it the onValueChange callback or maybe a ref to invoke this onChangeValue until we can implement this? It would be nice to have this feature

**Comment by MarlonAEC** [comment]
Btw @schmitzc I was able to kinda fake what you were trying to achieve (adding an icon to the selected one) like this:

```
// ... rest of my select component
<SelectContent padding="$6">
  <SelectDragIndicatorWrapper>
    <View
      width="$8"
      height="$8"
      backgroundColor="red"
      justifyContent="center"
      alignItems="center"
      alignSelf="flex-end"
    >
      <CloseIcon />
    </View>
  </SelectDragIndicatorWrapper>
  <SelectItem label="" value="" />
  {field?.options?.map((option) => {
    if (!option) {
      return null;
    }
    const isSelected = selectedValueForIcon === option.value;
    return (
      <View flexDirection="row">
        <SelectItem
          key={option.id}
          label={option.value}
          value={option.value}
          paddingHorizontal={0}
          flexWrap="wrap"
          width="90%"
        />
        {isSelected && (
          <Icon
            as={CheckCircleIcon}
            width="$6"
            height="$6"
            fill="$success300"
            stroke={colors.background}
            alignSelf="center"
          />
        )}
      </View>
    );
  })}
</SelectContent>
//... rest of my select component
```
notice there is variable called `selectedValueForIcon` is just a useState that I use to save the current selected value to verify if it's the current one selected and based on that I hide or show the icon I want. Hope that helps!
